### PR TITLE
Force auth users to complete profile

### PR DESCRIPTION
**Pull request by jimleeder123**
This PR is for where we are forcing authenticated users who haven't filled out their full details yet to be redirected to the profile page no matter what they do. The only exclusions from this are the sign out route and post sign out route as well as the actual profile and profile post routes (Causes a "too many redirects" error otherwise). 

There are questions in the comments as normal.

### CODE REVIEW

File: resources/views/web/profile/index.blade.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+@extends('skeleton')
+
+@section('page')
+
+    <h1>Update your profile</h1> {{-- todo: discuss - should this be update or edit? It's edit on the portal... --}}
+
+    @if($user->has_completed_profile)
+        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus nec dapibus lacus. In hac habitasse platea dictumst. Donec facilisis sem semper odio viverra, nec mattis felis sodales. Curabitur blandit, dui vel venenatis consequat, erat ligula semper orci, a scelerisque nisi risus feugiat ipsum. Integer molestie felis vel dignissim varius. Donec aliquet viverra scelerisque. Pellentesque eget congue mi. Duis tincidunt tortor orci, hendrerit dignissim ipsum facilisis ut. Praesent nec nulla non ligula pulvinar semper ut sed metus. Curabitur arcu lacus, vestibulum in molestie sed, sollicitudin eu lorem. Ut tempus et augue sed sollicitudin. Sed convallis nisl a felis gravida lobortis.</p>
+    @else
+        <p>You must complete your profile before using this website. Please fill out the form to proceed.</p>
```
Comment (gbuckingham89) [inline resources/views/web/profile/index.blade.php:10]: > To continue, you need to complete your profile. The fields marked with * are required.

**Comment by jimleeder123** [comment]
> Overall looks great, just a few small tweaks and I think we're good to go! Nice job capturing the infinite loop - and even more so for adding the tests for that.
> 
> Might worth adding 1 test, maybe for the portal index, that users are redirected to the edit profile page when they have an incomplete profile.

The test already exists on line 44 of the `Tests/Feature/Http/Web/PortalController/GetIndexTest` class. Perhaps I could rename it a bit though. But will get these other points done now, thanks.

File: app/Http/Middleware/EnforceAuthenticatedUsersCompleteProfile.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                [
+                    'web.profile.index',
+                    'web.profile.index.post',
+                    'web.auth.sign-out',
+                    'web.auth.sign-out.post',
+                ]
+            )
+        ) {
+            return redirect(route('web.profile.index'));
+        }
```
[the remark this answers is not in the document]
Comment (jimleeder123) [reply app/Http/Middleware/EnforceAuthenticatedUsersCompleteProfile.php:36]: This is better, thanks. The `Auth::check()` and `Auth::user()` was to stop Larastan marking it

File: app/Http/Middleware/EnforceAuthenticatedUsersCompleteProfile.php
[the remark this answers is not in the document]
Comment (jimleeder123) [reply app/Http/Middleware/EnforceAuthenticatedUsersCompleteProfile.php:1]: That is better, thanks

File: resources/views/web/profile/index.blade.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
+@extends('skeleton')
+
+@section('page')
+
+    <h1>Update your profile</h1> {{-- todo: discuss - should this be update or edit? It's edit on the portal... --}}
```
[the remark this answers is not in the document]
Comment (jimleeder123) [reply resources/views/web/profile/index.blade.php:5]: Done

File: resources/views/web/profile/index.blade.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            <x-forms.label for="first_name">First name *</x-forms.label>
+            <x-forms.text id="first_name" name="first_name" :value="old('first_name', $user->first_name)" />
+        </div>
+
+        <div>
+            <x-forms.label for="last_name">Last name *</x-forms.label>
+            <x-forms.text id="last_name" name="last_name" :value="old('last_name', $user->last_name)" />
+        </div>
+
+        <div> {{-- todo: discuss - should we allow editing of the email address or not? If we do then we may need more confirmation emails etc set up... --}}
```
[the remark this answers is not in the document]
Comment (jimleeder123) [reply resources/views/web/profile/index.blade.php:29]: Removed comment

File: resources/views/web/profile/index.blade.php (hunk 1 above)
Comment (jimleeder123) [reply resources/views/web/profile/index.blade.php:10]: Done

File: resources/views/web/profile/index.blade.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
+@extends('skeleton')
+
+@section('page')
+
+    <h1>Update your profile</h1> {{-- todo: discuss - should this be update or edit? It's edit on the portal... --}}
+
+    @if($user->has_completed_profile)
+        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus nec dapibus lacus. In hac habitasse platea dictumst. Donec facilisis sem semper odio viverra, nec mattis felis sodales. Curabitur blandit, dui vel venenatis consequat, erat ligula semper orci, a scelerisque nisi risus feugiat ipsum. Integer molestie felis vel dignissim varius. Donec aliquet viverra scelerisque. Pellentesque eget congue mi. Duis tincidunt tortor orci, hendrerit dignissim ipsum facilisis ut. Praesent nec nulla non ligula pulvinar semper ut sed metus. Curabitur arcu lacus, vestibulum in molestie sed, sollicitudin eu lorem. Ut tempus et augue sed sollicitudin. Sed convallis nisl a felis gravida lobortis.</p>
```
[the remark this answers is not in the document]
Comment (jimleeder123) [reply resources/views/web/profile/index.blade.php:8]: Done

File: resources/views/web/profile/index.blade.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            <x-forms.label for="phone">Phone</x-forms.label>
+            <x-forms.text type="tel" id="phone" name="phone" :value="old('phone', $user->phone)" />
+        </div>
+
+        <div>
+            <x-forms.checkbox name="marketing_opt_in" id="marketing_opt_in" :checked="old('marketing_opt_in', $user->marketing_opt_in)" value="1">I'm happy to receive marketing from Workplace Futures Group</x-forms.checkbox>
+        </div>
+
+        <div>
+            <button type="submit" title="Save changes">Save changes</button>
```
[the remark this answers is not in the document]
Comment (jimleeder123) [reply resources/views/web/profile/index.blade.php:49]: Done

**Review by gbuckingham89 — APPROVED** [review]
Looks good to go - nice one!

Apologies I missed the test already existed! 🙈

**Pull request merged**